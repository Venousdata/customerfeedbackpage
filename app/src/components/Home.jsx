import { CircularProgress, makeStyles } from "@material-ui/core";
import { useState, useEffect } from "react";
import { get } from "../api/api";
import Products from "./Products";

const useStyles = makeStyles((theme) => ({
  home: {
    textAlign: "center",
  },
}));

const Home = () => {
  const [products, setProducts] = useState([]);
  const classes = useStyles();

  useEffect(() => {
    get("products/all")
      .then((data) => setProducts(data))
      .catch((err) => console.log(err));
  }, []);

  return (
    <div className={classes.home}>
      <h2>Home</h2>
      <p>Check out some of our products!</p>
      {products ? <Products products={products} /> : <CircularProgress />}
    </div>
  );
};

export default Home;
