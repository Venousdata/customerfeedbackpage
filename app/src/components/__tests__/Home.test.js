import React from "react";
import { render } from "@testing-library/react";
import Home from "../Home";

describe("Home tests", () => {
  it("Home should render and match snapshot", (done) => {
    const container = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
    done();
  });
});
