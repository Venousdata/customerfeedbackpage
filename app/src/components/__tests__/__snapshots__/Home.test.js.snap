// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests Home should render and match snapshot 1`] = `undefined`;
